app/src/ui/history/file-list.tsx

```diff
@@ -66,4 +100,51 @@ export class FileList extends React.Component<IFileListProps, {}> {
       </div>
     )
   }
+
+  private onContextMenu = (event: React.MouseEvent<any>) => {
+    event.preventDefault()
+
+    if (this.props.selectedFile !== null) {
+      const filePath = this.props.selectedFile.path
+      const extension = Path.extname(this.props.selectedFile.path)
+      const items: IMenuItem[] = []
+
+      const isSafeExtension = __WIN32__
+        ? RestrictedFileExtensions.indexOf(extension.toLowerCase()) === -1
+        : true
+
+      const revealInFileManagerLabel = __DARWIN__
+        ? 'Reveal in Finder'
+        : __WIN32__ ? 'Show in Explorer' : 'Show in your File Manager'
+
+      const openInExternalEditor = this.props.externalEditorLabel
+        ? `Open in ${this.props.externalEditorLabel}`
+        : defaultEditorLabel
+
+      items.push(
+        {
+          label: revealInFileManagerLabel,
+          action: () => this.props.onRevealInFileManager(filePath),
+          enabled: this.props.selectedFile.status !== AppFileStatus.Deleted,
```

Comment:
This status is for when the commit was made - subsequent work might have deleted the file from disk, so this button could be enabled but the action might confuse the editor because there's nothing on disk.

What if we did a check like this?

```ts
const fullPath = Path.join(this.props.repository.path, filePath)
const fileExistsOnDisk = await pathExists(fullPath)
```

We don't have a `repository` prop yet, but that's easy enough to plumb through from the parent components, and [`pathExists`](https://github.com/desktop/desktop/blob/2dfc68ad0e5cc9f2fbd486e2dc3bd582176ee5f3/app/src/lib/file-system.ts#L109-L119) is function to check the underlying file system for whether this exists on disk.

This also means we can do a bit of tidy up in here:

```ts
      items.push(
        {
          label: revealInFileManagerLabel,
          action: () => this.props.onRevealInFileManager(filePath),
          enabled: fileExistsOnDisk, // this line is a bit easier to understand
        },
        ... 
```

Reply:
I have added requested feature in this [commit](https://github.com/desktop/desktop/pull/4037/commits/168d764c2f93ff2b26755d18ea00fdf426a3ecbf). However I'm not sure if I made correct replacement for fullPath in onOpenInExternalEditor and onOpenItem. onRevealInFileManager doesn't work with fullPath, so there is still filePath